src/popup/pages/SignMessage.vue

```diff
@@ -18,11 +18,14 @@ export default defineComponent({
 
       try {
         const aeSdk = await getAeSdk();
-        const message = route.query.message?.toString();
+        const rawMessage = route.query.message?.toString();
+        const isHexEncodedMessage = !!rawMessage && route.query.encoding?.toString() === 'hex';
+        const message = isHexEncodedMessage ? Buffer.from(rawMessage, 'hex') : rawMessage;
+        const displayMessage = message?.toString();
```

Comment:
Tiny suggestion:
```suggestion
        const message = (isHexEncodedMessage ? Buffer.from(rawMessage, 'hex') : rawMessage).toString();
```
This way you can still pass `message` to `openModal`.